src/Storefront/Page/Product/Configurator/ProductPageConfiguratorLoader.php

```diff
@@ -154,7 +154,12 @@ private function sortSettings(array $groups): PropertyGroupCollection
         usort(
             $sorted,
             static function (PropertyGroupEntity $a, PropertyGroupEntity $b) {
-                return strnatcmp($a->getTranslation('name'), $b->getTranslation('name'));
+                $posA = $a->getTranslation('position');
+                $posB = $b->getTranslation('position');
+                if ($posA === $posB) {
+                    return strnatcmp($a->getTranslation('name'), $b->getTranslation('name'));
+                }
+                return ($posA < $posB) ? -1 : 1;
```

Comment:
As of php 7.0 there is a spaceship operator. 
`return $posA <=> $posB;`

Reply:
TIL, thanks.